CMPUT301Project/app/src/main/java/com/example/GroupProject/HabitActivityHelper.java

```diff
@@ -33,63 +39,48 @@
      * @param day Integer day to check.
      * @return true if it is a valid day, false otherwise.
      */
-    public static Boolean isDateToStartGood(Integer year, Integer month, Integer day) {
+    public static Boolean isDateToStartGood(Long year, Long month, Long day) {
         Boolean dayGood = false;
-        switch (month) {
-            // JAN
-            case 0:
-                dayGood = day <= 31;
-                break;
-            // FEB
-            case 1:
-                if (year % 4 == 0){
-                    dayGood = day <= 29;
-                } else {
-                    dayGood = day <= 28;
-                }
-                break;
-            // MAR
-            case 2:
-                dayGood = day <= 31;
-                break;
-            // APR
-            case 3:
-                dayGood = day <= 30;
-                break;
-            // MAY
-            case 4:
-                dayGood = day <= 31;
-                break;
-            // JUN
-            case 5:
-                dayGood = day <= 30;
-                break;
-            // JUL
-            case 6:
-                dayGood = day <= 31;
-                break;
-            // AUG
-            case 7:
-                dayGood = day <= 31;
-                break;
-            // SEP
-            case 8:
-                dayGood = day <= 30;
-                break;
-            // OCT
-            case 9:
-                dayGood = day <= 31;
-                break;
-            // NOV
-            case 10:
-                dayGood = day <= 30;
-                break;
-            // DEC
-            case 11:
-                dayGood = day <= 31;
-                break;
-            default:
-                break;
+        // JAN
+        if (month == 1) {
```

Comment:
This can be compacted into fewer if/switch/case statement since many of the cases are the same. Just letting you know for next time.

Reply:
It warned me about that but thought it was more clear broken up. Less compact but more readable imo. Thank you though.